recipe/build.sh

```diff
@@ -8,16 +8,12 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
   # Set the CUDA arch list from
   # https://github.com/conda-forge/pytorch-cpu-feedstock/blob/86592106fc0e4731eb011ac763a4f0429326930b/recipe/build_pytorch.sh#L123
 
-  if [[ ${cuda_compiler_version} == 10.* ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5+PTX"
-  elif [[ ${cuda_compiler_version} == 11.0* ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0+PTX"
-  elif [[ ${cuda_compiler_version} == 11.1 ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6+PTX"
-  elif [[ ${cuda_compiler_version} == 11.2 ]]; then
+  if [[ ${cuda_compiler_version} == 11.2 ]]; then
     export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6+PTX"
   elif [[ ${cuda_compiler_version} == 11.8 ]]; then
     export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+  elif [[ ${cuda_compiler_version} == 12.0 ]]; then
+    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
```

Comment:
Getting a mamba solver error at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=818010&view=logs&j=81eb4d60-76fc-5ac4-a959-9ebb9871bfee&t=cacffbe7-e028-5a90-8f74-056b2f489f53&l=70:

```python-traceback
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 141, in mamba_get_install_actions
    solution = solver.solve_for_action(_specs, prefix)
  File "/opt/conda/lib/python3.10/site-packages/boa/core/solver.py", line 230, in solve_for_action
    t = self.solve(specs)
  File "/opt/conda/lib/python3.10/site-packages/boa/core/solver.py", line 220, in solve
    raise RuntimeError("Solver could not find solution." + error_string)
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - git
 - cuda-version >=12.0,<13
 - libaio
 - pytorch * cuda*
 - pip
 - libstdcxx-ng >=12
 - pytorch 2.0.*
 - libgcc-ng >=12
 - py-cpuinfo
 - python 3.11.* *_cpython

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package pytorch-1.6.0-cuda100py36hd82b6f9_1 requires cudatoolkit 10.0|10.0.*, but none of the providers can be installed
```

Need to wait for CUDA 12 builds of Pytorch to be uploaded to conda-forge channel. PR seems to have been merged already (at https://github.com/conda-forge/pytorch-cpu-feedstock/pull/192?), but the build timed out.